compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -198,7 +198,7 @@ crate struct CrateRoot<'tcx> {
     extra_filename: String,
     hash: Svh,
     stable_crate_id: StableCrateId,
-    panic_strategy: PanicStrategy,
+    panic_strategy: Option<PanicStrategy>,
```

Comment:
What would cause this to be `None`?

Reply:
If the crate is compiled with `-C panic=unwind` and contains no FFI unwind calls. I.e. it does not require a particular panic runtime to be linked.

Reply:
Got it, thanks!

Reply:
That seems a bit odd? The panic strategy still makes a different for other things, doesn't it?

Reply:
This field should definitely be renamed (perhaps `requires_panic_runtime`?), it is very confusing at the moment since "panic strategy" refers to how the current crate was compiled.

Reply:
`#![no_std]` code doesn't require a panic runtime, but could still require a particular panic strategy for any dependent crates. For example a `-Cpanic=abort` no std crate requires all dependent crates to be compiled with `-Cpanic=abort` too, even though using `#[panic_handler]` rather than linking a panic runtime would work.

Reply:
This field and related query (`panic_strategy(_: CrateNum)`) is only used for checking compatibility. I could rename this to `required_panic_strategy`.